component-selenium-grid.yml

```diff
@@ -1,104 +1,264 @@
+     remote-url: bind(selenium-hub#result.remote-url)
     hosts:
-      hub: bind(selenium-grid#result.hub-host)
-      nodes: bind(selenium-grid#result.node-hosts)
-    actions:
-      scale: bind(selenium-grid#actions.scale)
-      re-start: bind(selenium-grid#actions.re-start)
-      change-version: bind(selenium-grid#actions.change-version)
-
+      hub: bind(selenium-hub#result.hub-host)
+      #nodes: bind(selenium-grid#result.node-hosts)
+  bindings:
+    - [linux-nodes.selenium-linux-node, selenium-hub]
+    - [windows-nodes.selenium-windows-node, selenium-hub]
+    - [proxy#linux-nodes, linux-nodes]
+    - [proxy#windows-nodes, windows-nodes]
   components:
-    selenium-grid:
+    metadata:
+      type: cobalt.common.Constants
+      interfaces:
+        output:
+          application-pic:
+            type: publish-signal(map<string, object>)
+            name: ""
+      configuration:
+        configuration.values:
+          output.application-pic:
+            large: "https://s3.amazonaws.com/qubell-images/selenium-logo.png"
+            small: "https://s3.amazonaws.com/qubell-images/selenium-logo-small.png"
+            small-height: 64
+    selenium-hub:
       type: workflow.Instance
       interfaces:
         input:
-          selenium-version:
-            type: configuration(string)
-            name: Selenium Version
-          node-count:
-            type: configuration(string)
-            name: Count of Additional Nodes
-          recipes:
-            type: configuration(string)
-            name: Recipes Url
+          hub-image:
+            type: configuration(map<string, object>)
+            suggestions:
+              "Win2008r2 (us-east-1)": { ami: "us-east-1/ami-31620c54", user: "Administrator", type: "windows", hw: "m3.large", issudo: false, prefix: "Selenium-hub"  }
+              "Win2008r2 (us-west-1)": { ami: "us-west-1/ami-2b2dd76f", user: "Administrator", type: "windows", hw: "m3.large", issudo: false, prefix: "Selenium-hub"  }
+              "CentOS 7.1 (us-east-1)": { ami: "us-east-1/ami-96a818fe", user: "centos", type: "linux", hw: "m3.medium", issudo: true, prefix: "Selenium-hub" }
+              "CentOS 7.1 (us-west-1)": { ami: "us-west-1/ami-6bcfc42e", user: "centos", type: "linux", hw: "m3.medium", issudo: true, prefix: "Selenium-hub" }
+          selenium-version: { type: configuration(string), name: Selenium Version }
+          ie-driver-version: { type: configuration(string), name: IE Driver Version }
+          chrome-driver-version: { type: configuration(string), name: Chrome Driver Version }
+          recipes: { type: configuration(string), name: Recipes Url }
+          java-win-url: { type: configuration(string), name: "Windows Java installer URL" }
+          windows-user-password: { type: configuration(string), name: "Windows Administrator user password" }
+          mirror: { type: configuration(string), name: "Cygwin installation binary URL" }
```

Comment:
We use it everywhere. You'd prefer cygwin-url? Does it make any difference as name exposed like "Cygwin installation binary URL"?